src/app/shared/pipes/filter.pipe.ts

```diff
@@ -1,26 +1,26 @@
 import { Pipe, PipeTransform } from "@angular/core";
 
-import { FilterType } from "@shared/models/types/filter-type.type";
+import { ActiveFilters } from "@shared/models/types/active-filters.type";
 
 @Pipe({
     name: "filter"
 })
 export class FilterPipe implements PipeTransform {
 
-    transform(items: Object[], filter: FilterType): Object[] {
+    transform<T>(items?: T[], filter?: ActiveFilters): T[] {
         if (items == null) { return []; }
-        if (filter == null) { return items; }
+        if (filter == null || Object.keys(filter).length == 0) { return items; }
 
         return items.filter(item => this.includedByFilter(filter, item));
     }
 
-    includedByFilter(filter: FilterType, item: Object): boolean {
+    includedByFilter(filter: ActiveFilters, item: Object): boolean {
         return Object.entries(item)
             .some(([k, v]) => this.matchingValue(filter, k, v));
     }
 
-    matchingValue(filter: FilterType, key: string, value: string): boolean {
+    matchingValue(filter: ActiveFilters, key: string, value: string): boolean {
         return Object.keys(filter).includes(key)
-            && filter[key].subOptions.includes(value);
+            && filter[key].includes(String(value));
```

Comment:
For some reason, although `value` is of type `string`, when the prices were getting passed in (also of type `string` on the `Product` class), they were getting converted to `number` types, so the `includes` call was then failing as, for example, `"12.25" == 12.25` is `false`. Explicitly converting to `string` fixed this, but it doesn't feel like this should be necessary.

Reply:
that's fine for now, we can always convert price to a number in the class, might be easier